File: site/mp2i/files/C7/correction.c
# include <stdio.h>

/* Renvoie le nombre d'occurrence de elt dans tab */
int nb_occurence(int elt, int tab[], int size)
{
    int nb = 0;
    for (int i=0; i<size; i++)
    {
        if (tab[i]==elt)
        {
            nb = nb + 1;
        }
    }
    return nb;
}

int main()
{
    int tab[7] = {4, 7, 5, 3, 4, 1, 4};
    printf("Occurences  = %d\n",nb_occurence(10,tab,7));
}

File: itc/files/C16/correction.py
# Renvoie le nombre d'occurences d'elt dans liste
def nb_occ(elt, liste):
    cpt = 0
    for x in liste:
        if x == elt:
            cpt = cpt + 1
    return cpt

# Calcule n!
def fact(n):
    if n==0:
        return 1
    return n * fact(n-1)

# Renvoie la liste en dupliquant chaque élément
def duplique(liste):
    if liste==[]:
        return []
    return [liste[0],liste[0]] + duplique(liste[1:])